Customer feedback: add profile inspector foldouts

davidkline-ms opened the pull request:
Overview
---
Per customer feedback, adds foldouts to the following profile inspectors:

* Mixed Reality Toolkit Configuration
* Boundary Visualization
* Camera
* Diagnostics
* Spatial Awareness
* Speech

Changes
---
- Fixes: #3215 .
Plus a few minor cleanup changes (ex: replace GUILayoyt.Space(12f) with EditorGUILayout.Space() [ Same visuals, easier to read ] and add a comment.

StephenHodgson commented:
Can we get some images of what these changes look like?

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealitySpeechCommandsProfileInspector.cs:
```diff
@@ -86,7 +87,13 @@ public override void OnInspectorGUI()
 
-            RenderList(speechCommands);
+            EditorGUILayout.Space();
+            showSpeechCommands = EditorGUILayout.Foldout(showSpeechCommands, "Speech Commands");
```
Why do we need a foldout for this?

It's literally the only thing in this profile. I think it is a bit overboard.

keveleigh replied:
It's not about the amount of content but the prevention of context switching.

davidkline-ms replied:
> None of the other input profiles with a similar list was updated.

The speech profile was the first that got changed. The others are in the backlog.

The reason for enabling folding this list is to allow customers to simplify their view when not adding new commands.

davidkline-ms commented:
Here are some examples of the change. I am reviewing the customer feedback to ensure the implementation addresses the concerns.

![capture](https://user-images.githubusercontent.com/13281406/49535630-799f1480-f879-11e8-9638-6bedf66ff2e9.PNG)

![capture1](https://user-images.githubusercontent.com/13281406/49535635-7c9a0500-f879-11e8-86e4-6b079fc931df.PNG)

StephenHodgson commented:
Looks, okay, but I suggest we also indent the sections hidden by the foldouts.

StephenHodgson commented:
I think we should only add the foldouts if the inspector goes off the bottom of the page.

davidkline-ms commented:
> I think we should only add the foldouts if the inspector goes off the bottom of the page.

On further review of the feedback, this PR is likely to change. The customer request is for our inspector to behave like the hierarchy and allow them to show only the data of interest.

I expect to have a revision early next week that may illustrate the feedback better,

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityBoundaryVisualizationProfileInspector.cs:
```diff
@@ -90,34 +96,50 @@ public override void OnInspectorGUI()
             EditorGUILayout.PropertyField(boundaryHeight);
+
+            EditorGUILayout.Space();
+            showFloorProperties = EditorGUILayout.Foldout(showFloorProperties, "Floor Settings");
```
```suggestion
            showFloorProperties = EditorGUILayout.Foldout(showFloorProperties, "Floor Settings", true);
```

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityBoundaryVisualizationProfileInspector.cs:
```diff
@@ -90,34 +96,50 @@ public override void OnInspectorGUI()
+            }
+
+            EditorGUILayout.Space();
+            showPlayAreaProperties = EditorGUILayout.Foldout(showPlayAreaProperties, "Play Area Settings");
```
```suggestion
            showPlayAreaProperties = EditorGUILayout.Foldout(showPlayAreaProperties, "Play Area Settings", true);
```

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityBoundaryVisualizationProfileInspector.cs:
```diff
@@ -90,34 +96,50 @@ public override void OnInspectorGUI()
-            EditorGUILayout.LabelField("Boundary Ceiling Settings:", EditorStyles.boldLabel);
-            EditorGUILayout.PropertyField(showBoundaryCeiling, showContent);
-            EditorGUILayout.PropertyField(boundaryCeilingMaterial, materialContent);
+            showTrackedAreaProperties = EditorGUILayout.Foldout(showTrackedAreaProperties, "Tracked Area Settings");
```
```suggestion
            showTrackedAreaProperties = EditorGUILayout.Foldout(showTrackedAreaProperties, "Tracked Area Settings", true);
```

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityBoundaryVisualizationProfileInspector.cs:
```diff
@@ -90,34 +96,50 @@ public override void OnInspectorGUI()
+            }
+
+            EditorGUILayout.Space();
+            showWallProperties = EditorGUILayout.Foldout(showWallProperties, "Boundary Wall Settings");
```
```suggestion
            showWallProperties = EditorGUILayout.Foldout(showWallProperties, "Boundary Wall Settings", true);
```

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityBoundaryVisualizationProfileInspector.cs:
```diff
@@ -90,34 +96,50 @@ public override void OnInspectorGUI()
+            }
+
+            EditorGUILayout.Space();
+            showCeilingProperties = EditorGUILayout.Foldout(showCeilingProperties, "Boundary Ceiling Settings");
```
```suggestion
            showCeilingProperties = EditorGUILayout.Foldout(showCeilingProperties, "Boundary Ceiling Settings", true);
```

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityCameraProfileInspector.cs:
```diff
@@ -68,32 +72,41 @@ public override void OnInspectorGUI()
-            EditorGUILayout.PropertyField(opaqueClearFlags, clearFlagsTitle);
-
-            if ((CameraClearFlags)opaqueClearFlags.intValue == CameraClearFlags.Color)
+            showGlobalProperties = EditorGUILayout.Foldout(showGlobalProperties, "Global Settings");
```
```suggestion
            showGlobalProperties = EditorGUILayout.Foldout(showGlobalProperties, "Global Settings", true);
```

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityCameraProfileInspector.cs:
```diff
@@ -68,32 +72,41 @@ public override void OnInspectorGUI()
             }
+            
+            EditorGUILayout.Space();
+            showOpaqueProperties = EditorGUILayout.Foldout(showOpaqueProperties, "Opaque Display Settings");
```
```suggestion
            showOpaqueProperties = EditorGUILayout.Foldout(showOpaqueProperties, "Opaque Display Settings", true);
```

StephenHodgson commented on Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityDiagnosticsSystemProfileInspector.cs:
```diff
@@ -71,27 +71,36 @@ public override void OnInspectorGUI()
             {
-                EditorGUILayout.PropertyField(showCpu);
-                EditorGUILayout.PropertyField(cpuBuffer);
+                using (new EditorGUI.IndentLevelScope())
```
Why not just use the `EditorGUI.indentLevel==` and save the allocations with having a new one?

We're already in brackets for the foldout anyway, so seems rather wasteful.

davidkline-ms replied:
I originally chose the using pattern to prevent forgetting to unindent (defensive coding). The allocations would only occur in the editor, so should not impact on device performance.

davidkline-ms commented:
The latest commits update to current mrtk_development and make the profile labels more consistent (ex: general/global system settings are now labeled as General).

This change does not address the following inspectors (they will be covered in a future PR):
- Controller*ProfileInspector
- Input*ProfileInspector
- GesturesProfileInspector
